torchbox/torchbox.com
# Override page titles in the navigation

[Link to Ticket](https://torchbox.monday.com/boards/1192293412/pulses/1472012193)

### Description of Changes Made

- Fix some validation issues and field names in the navigation settings
    - Don't require title text for external links for the footer logos
    - Use correct help text for the title field in the primary nav links and footer links
    - Add an `alt text` field for footer logos, ensure this is used in the template, and update the pattern library template. Note that there is no data migration here and the alt text needs to be filled out to match the current title text after deployment.
- Add a new mixin class in utils that allows a `navigation_text` field to be added to page models.
- Update all page models to inherit from this mixin, and ensure it is added to the promote panel
- Update all relevant navigation templates and yaml files

Note that for consistency I have added the new field to all page models, but in reality, given the way that the navigation is set up, some page types will never use it - work, events and blog page models are all either top level navigation items, or are not shown in the navigation.

### How to Test

- Pull production data
- Edit a service page immediately below "services", e.g. "Digital marketing", and enter a different value in the 'navigation text' field. Check that the new text appears in the menu both at desktop and mobile.
- Edit a service page at the next level down, e.g. "Digital marketing training for charities", and enter a different value in the 'navigation text' field. Check that the new text appears in the menu both at desktop and mobile.
- Edit a page under "About", e.g. "Team". Check that the new text appears in the menu both at desktop and mobile.
- Check that the navigation renders in the pattern library as expected at http://localhost:8000/pattern-library/render-pattern/patterns/organisms/header/header.html.

### Screenshots

<details>
  <summary>The new navigation text field when editing a page</summary>

![Screenshot 2024-04-22 at 13 12 00](https://github.com/torchbox/torchbox.com/assets/771869/b47c2619-aef2-4bad-9877-b5965ef3ee39)

</details>

<details>
  <summary>Setting the navigation text at the top level - note the corrected help text</summary>

![Screenshot 2024-04-22 at 13 13 06](https://github.com/torchbox/torchbox.com/assets/771869/dfec6d66-1b94-4065-8e00-0b27ef9249f9)

</details>

<details>
  <summary>The footer logo link in the navigation settings with corrected field name and validation</summary>

![Screenshot 2024-04-22 at 13 13 59](https://github.com/torchbox/torchbox.com/assets/771869/06431115-8b96-420b-8895-b9505beb0f0c)

</details>

### MR Checklist


#### Unit tests


#### Documentation


#### Browser testing

  - Latest version of Chrome on mac

#### Data protection


#### Accessibility


#### Sustainability


#### Pattern library
